src/app/actions/updateProductAction.ts

```diff
@@ -1,40 +1,184 @@
 'use server';
 
-import { doc, updateDoc } from 'firebase/firestore';
-import { db, collections } from '@/utils/firebase';
-import { z } from 'zod';
+import { doc, getDoc, updateDoc } from "firebase/firestore";
+import { db, collections } from "@/utils/firebase";
+import { stripe } from "@/utils/stripe";
+import { z } from "zod";
+import type { Product } from "@/types/product";
 
-const productSchema = z.object({
-  id: z.string(),
+const updateSchema = z.object({
+  id: z.string().min(1, "Missing product id"),
   title: z.string().min(3).max(100),
-  category: z.string(),
+  category: z.string().min(1),
   price: z.number().min(0),
   stock: z.number().min(0),
 });
 
-export async function updateProductAction(formData: FormData) {
-  const rawData = {
-    id: formData.get('id') as string,
-    title: formData.get('title') as string,
-    category: formData.get('category') as string,
-    price: parseFloat(formData.get('price') as string),
-    stock: parseInt(formData.get('stock') as string, 10),
-  };
+type ActionOk = { success: true; message: string; stripePriceId?: string };
+type ActionErr = {
+  success: false;
+  message: string;
+  issues?: { path: string; message: string }[];
+};
+type ActionResult = ActionOk | ActionErr;
 
-  const result = productSchema.safeParse(rawData);
+const num = (v: FormDataEntryValue | null) =>
+  v == null || v === "" ? undefined : Number(v);
 
-  if (!result.success) {
-    return { success: false, message: 'Validation failed', errors: result.error.flatten().fieldErrors };
-  }
+const isNumber = (n: unknown): n is number =>
+  typeof n === "number" && Number.isFinite(n);
+
+const zodIssues = (err: z.ZodError): { path: string; message: string }[] =>
+  err.issues.map(i => ({
+    path: i.path.join("."),
+    message: i.message,
+  }));
+
+export async function updateProductAction(formData: FormData): Promise<ActionResult> {
+  try {
+    
+    const raw = {
+      id: String(formData.get("id") || ""),
+      title: String(formData.get("title") || ""),
+      category: String(formData.get("category") || ""),
+      price: Number(formData.get("price")),
+      stock: Number(formData.get("stock")),
+      description: formData.get("description")?.toString(),
+      brand: formData.get("brand")?.toString(),
+      sku: formData.get("sku")?.toString(),
+      weight: num(formData.get("weight")),
+      warrantyInformation: formData.get("warrantyInformation")?.toString(),
+      shippingInformation: formData.get("shippingInformation")?.toString(),
+      availabilityStatus: formData.get("availabilityStatus")?.toString(),
+      returnPolicy: formData.get("returnPolicy")?.toString(),
+      dimW: num(formData.get("dimensions.width")),
```

Comment:
Is there any reason to have dimentions as dimW, dimH, dimD? 
It's usually better to keep data in the same format if possible: 
```JS
dimensions: z.object({
    width: z.number().min(1),
    height: z.number().min(1),
    depth: z.number().min(1),
  }),
```